sql/table.go

```diff
@@ -164,6 +164,8 @@ func encodeTableKey(b []byte, v parser.Datum) ([]byte, error) {
 		return encoding.EncodeNumericFloat(b, float64(t)), nil
 	case parser.DString:
 		return encoding.EncodeBytes(b, []byte(t)), nil
+	case parser.DNull:
+		return b, nil
```

Comment:
Indexes can often be used for ordering in order to avoid an explicit sort. See the "sorting by index" section in https://www.sqlite.org/queryplanner.html for some examples.

I don't think we need to support both NULLs first and last, but we probably do want to choose one rather than letting NULLs sort anywhere or requiring an explicit in-memory sort when NULLs are present.